homeassistant/helpers/event.py

```diff
@@ -167,6 +167,52 @@ def state_for_cancel_listener(entity, from_state, to_state):
 track_same_state = threaded_listener_factory(async_track_same_state)
 
 
+@callback
+@bind_hass
+def async_track_same_template(hass, period, template, action, variables=None,
```

Comment:
Why make a new function? `async_track_same_state` allows passing in a function.

Reply:
Hmm, I guess I didn't think of that. I'll give that a shot. Thanks!

Reply:
I think it still makes sense to add `async_track_same_template`, otherwise details of how these functions in helpers/event.py will leak into components/automation/template.py. Things would become unbalanced. But it _does_ make sense to reuse `async_track_same_state` in the implementation of `async_track_same_template`, making it _much_ simpler.

Reply:
I suggest we start with adding that definition to `automation/template.py`, and we can always move it to `helpers/event.py` if more use cases appear.

Reply:
Done.